src/main/java/com/github/checkstyle/regression/module/ModuleUtils.java

```diff
@@ -0,0 +1,133 @@
+//
+// This library is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+// Lesser General Public License for more details.
+//
+// You should have received a copy of the GNU Lesser General Public
+// License along with this library; if not, write to the Free Software
+// Foundation, Inc., 59 Temple Place, Suite 330, Boston, MA  02111-1307  USA
+////////////////////////////////////////////////////////////////////////////////
+
+package com.github.checkstyle.regression.module;
+
+import java.io.IOException;
+import java.io.InputStream;
+import java.io.InputStreamReader;
+import java.nio.charset.Charset;
+import java.util.Arrays;
+import java.util.List;
+import java.util.Map;
+import java.util.regex.Pattern;
+import java.util.stream.Collectors;
+
+import com.github.checkstyle.regression.data.GitChange;
+import com.github.checkstyle.regression.util.GsonProvider;
+import com.google.gson.Gson;
+import com.google.gson.reflect.TypeToken;
+
+/**
+ * Contains utility methods related to checkstyle module.
+ * @author LuoLiangchen
+ */
+public final class ModuleUtils {
+    /** The compiled regex pattern of the path of Java main source files. */
+    private static final Pattern MAIN_SOURCE_PATTERN = Pattern.compile("src/main/java/(.+)\\.java");
+
+    /** The full qualified name to module info map. */
+    private static final Map<String, ModuleInfo> NAME_TO_MODULE_INFO;
+
+    /** Prevents instantiation. */
+    private ModuleUtils() {
+    }
+
+    static {
+        NAME_TO_MODULE_INFO = createNameToModuleInfo();
+    }
+
+    /**
+     * Creates the full qualified name to module info map.
+     * @return the full qualified name to module info map
+     */
+    private static Map<String, ModuleInfo> createNameToModuleInfo() {
+        final InputStream stream =
+                ModuleUtils.class.getClassLoader().getResourceAsStream("checkstyle_modules.json");
+        return getNameToModuleInfoFromInputStream(stream);
+    }
+
+    /**
+     * Gets the full qualified name to module info map from the give input stream.
+     * @param stream the give input stream
+     * @return the full qualified name to module info map
+     */
+    private static Map<String, ModuleInfo> getNameToModuleInfoFromInputStream(InputStream stream) {
+        final List<ModuleInfo> modules;
+        final Gson gson = GsonProvider.provideGson();
+
+        try {
+            final InputStreamReader reader =
+                    new InputStreamReader(stream, Charset.forName("UTF-8"));
+
+            try {
+                modules = gson.fromJson(reader, new TypeToken<List<ModuleInfo>>() {
+                }.getType());
+            }
+            finally {
+                reader.close();
+            }
+        }
+        catch (IOException ex) {
+            throw new IllegalStateException("Failed when loaing hardcode information", ex);
```

Comment:
`hardcode information` => `hardcode checkstyle module information`
Best to be as specific as possible.